darshan-runtime/lib/darshan-heatmap.c

```diff
@@ -419,13 +420,25 @@ void heatmap_update(darshan_record_id heatmap_id, int rw_flag,
             return;
         }
 
+        intermediate_bytes = round(size * (seconds_in_bin/(end_time-start_time)));
+
+        /* NOTE: intermediate_bytes consistently produced the int64_t value
+         * -9223372036854775808 in the early stages of monitoring
+         *  for the case of round(0 * (0 / 0)), so just set to 0 when negative;
+         *  this appears to happen primarily in the very early stages
+         *  of monitoring when seconds_in_bin is 0.0; see gh-730
+         *  and https://github.com/darshan-hpc/darshan-logs/pull/35
+         */
+        if(intermediate_bytes < 0)
+            intermediate_bytes = 0;
```

Comment:
it also occurs to me that short-circuiting to `0` when `size` is `0` might save a bit of electricity